core/src/components/datetime/datetime.tsx

```diff
@@ -1415,7 +1518,11 @@ export class Datetime implements ComponentInterface {
 
     if (isDateEnabled) {
       days = days.map((dayObject) => {
-        const referenceParts = { month: workingParts.month, day: dayObject.value, year: workingParts.year };
+        const referenceParts = {
+          month: workingParts.month,
+          day: dayObject.value,
+          year: workingParts.year,
+        } as DatetimeParts;
```

Comment:
Do we cast after the value since `referenceParts` doesn't satisfy the full type signature?
i.e.: 
```ts
const referenceParts: DatetimeParts = {};
// vs.
const referenceParts = {} as DatetimeParts;
```